File: robocode2-bot-api-java/src/main/java/net/robocode2/events/GameEndedEvent.java
package net.robocode2.events;

import lombok.Builder;
import lombok.ToString;
import lombok.Value;
import net.robocode2.BotResults;

import java.util.List;

/** Event occurring when game has just ended */
@Value
@ToString
public class GameEndedEvent implements Message {
  /** Number of rounds played */
  int numberOfRounds;
  /** Results of the battle */
  List<BotResults> results;

  @Builder
  private GameEndedEvent(int numberOfRounds, List<BotResults> results) {
    this.numberOfRounds = numberOfRounds;
    this.results = results;
  }
}

File: robocode-tankroyale-bot-api-dotnet-core/events/GameEndedEvent.cs
using System.Collections.Generic;

namespace Robocode.TankRoyale
{
  /// <summary>
  /// Event occurring when game has just ended.
  /// </summary>
  public class GameEndedEvent : IMessage
  {
    /// <summary>Number of rounds played.</summary>
    int NumberOfRounds { get; }

    /// <summary>Results of the battle.</summary>
    List<BotResults> Results { get; }

    /// <summary>
    /// Constructor.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    /// <param name="victimId">ID of the bot that has died.</param>
    public GameEndedEvent(int numberOfRounds, List<BotResults> results) : base() =>
      (NumberOfRounds, Results) = (numberOfRounds, results);
  }
}